src/Storefront/Resources/app/storefront/src/plugin/listing/listing.plugin.js

```diff
@@ -252,7 +251,7 @@ export default class ListingPlugin extends Plugin {
     }
 
     _updateHistory(query) {
-        HistoryUtil.push(HistoryUtil.getLocation().pathname, query, {});
+        window.history.pushState({}, '', `${window.location.pathname}?${query}`);
```

Comment:
I am not the JS guy, as you know, but shouldn't we fix our util instead? :thinking:

Reply:
To be honest I want to get rid of it. 
What are your thoughts @tobiasberge?

Reply:
Would be fine for me, some functionality (pushParams, replaceParams) is also unused, at least in shopware itself. If we deprecate it I would do it for 6.8 to avoid more breaks. 

Stuff like "replace" also seems like an unneeded abstraction to me and the parameters get passed 1-to-1:

```js
replace(pathname, search, state) {
    this._history.replace({ pathname, search, state });
}
```

The stuff in `createBrowserHistory` (remix history) might need some more boilerplate afterwards, but if its mostly 1-liners or 2-liners I would prefer using the native APIs directly instead of inventing tooling around it.

We should double-check what https://github.com/browserstate/history.js is doing and if we really need it.